### PR TITLE
Added Recipe for removing Empty XML tags

### LINKED ISSUE #2737: XML - Remove empty tags
Provide a generic recipe to remove empty tags from a given XML file. 

This could be useful for changes like #2736 where the recipe leaves the pom file with an empty block that is no longer needed

Example : 

```xml
<project>
<groupId>com.mycompany.app</groupId>
<artifactId>my-app</artifactId>
<version>1</version>
<pluginRepositories>
</pluginRepositories>
</project>
```

```xml
<project>
<groupId>com.mycompany.app</groupId>
<artifactId>my-app</artifactId>
<version>1</version>
</project>
```

### PR DESCRIPTION
**Pull request by TheMarvelFan**
Closes #2737
Added the recipe to check for and remove empty tags with no children. 
Also added a test for this recipe.
Working on configuring this recipe.

### CODE REVIEW

File: rewrite-xml/build.gradle.kts
Comment (timtebeek) [inline rewrite-xml/build.gradle.kts:1]: It looks like you had some issues setting up this project in your IDE, and had to make changes to accommodate. Did you already go over [our IntelliJ IDEA setup instructions](https://github.com/openrewrite/.github/blob/main/CONTRIBUTING.md#building--developing)?

Perhaps best to revert these changes; I think they ought not be necessary for the recipe you're developing.

Comment (TheMarvelFan) [reply rewrite-xml/build.gradle.kts:1]: Yes, in fact I am facing some issues on the IDE regarding the gradle build. Let me provide a detailed overview:
IDE: IntelliJ Idea Ultimate 2023.2.3
Gradle Version: 8.4 complete zip
When I start the project on IntelliJ, I get the following error: 
"Could not create parent directory for lock file C:\Program Files\Gradle\gradle-8.4\bin\wrapper\dists\gradle-8.4-bin\1w5dpkrfk8irigvoxmyhowfim\gradle-8.4-bin.zip.lck"
Before this I got a different error saying the Class Compiler version was not supported (required: 55) so I set my java home to the jdk-11 directory, and it was solved.
But this has been really persistent, and I could not find anything in the documentation either. 
I will try to see if I missed something from the IntelliJ setup instructions, but I am pretty sure I covered everything.
PS: I will revert the changes in the build.gradle.kts file.

File: rewrite-xml/src/test/java/org/openrewrite/xml/RemoveEmptyXmlTagsRecipeTest.java [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+        rewriteRun(
+          spec -> spec.recipe(new RemoveEmptyXmlTagsRecipe()),
+          xml( """
+            <project>
+                <modelVersion>4.0.0</modelVersion>
+                <groupId>com.example</groupId>
+                <artifactId>sample-app</artifactId>
+                <version>1.0-SNAPSHOT</version>
+                <pluginRepositories>
+                </pluginRepositories>
```
Comment (timtebeek) [inline rewrite-xml/src/test/java/org/openrewrite/xml/RemoveEmptyXmlTagsRecipeTest.java:24]: This currently covers one of the scenarios listed in [this earlier comment](https://github.com/openrewrite/rewrite/issues/2737#issuecomment-1755732686); could we add two more tests for those alternatives just so we don't have any surprises when we run this recipe?

File: rewrite-xml/src/main/java/org/openrewrite/xml/tree/Xml.java [hunk 2]
```diff
@@ -465,6 +465,13 @@ public String getPrefix() {
             public String toString() {
                 return "</" + name + ">";
             }
+
+            public boolean isSelfClosing(Xml.Tag tag) {
```
Comment (timtebeek) [inline rewrite-xml/src/main/java/org/openrewrite/xml/tree/Xml.java:469]: Given how small this method is, and for now only used in the visitor, I'd pull this method to the visitor instead. That helps limit our publicly visible API & what we have to support going forward.

**Comment by timtebeek** [comment]
Given what we have here now; do we then support [the scenario described here](https://github.com/openrewrite/rewrite/issues/2737#issuecomment-1408520438) with the existing visitor? The goal is to tag-on a visitor when we for instance remove a `<plugin>`, and then walk up the parent hierarchy to remove any empty `<plugins>` and `<build>` we might encounter.

**Review by timtebeek — COMMENTED** [review]
Great work so far! A few comments/questions before we can merge, but otherwise solid effort, thanks!

**Comment by timtebeek** [comment]
Had another look; we already had a `RemoveContentVisitor` which was able to remove parents; I've updated that to now also look at attributes, just to be sure we don't remove too eagerly. That makes this PR now mostly added tests where behavior wasn't yet explicitly covered by tests, and a new recipe that folks can run when they are aware of the effects.

Thanks for getting this started @TheMarvelFan !

**Comment by TheMarvelFan** [comment]
Hi @timtebeek 
I apologize for not being able to follow through on this pull request. I was busy with preparing my project for the MLH fellowship. I will be able to start contributing again by tomorrow. 
PS: I was really enjoying working on this issue.

**Comment by timtebeek** [comment]
> Hi @timtebeek I apologize for not being able to follow through on this pull request. I was busy with preparing my project for the MLH fellowship. I will be able to start contributing again by tomorrow. PS: I was really enjoying working on this issue.

No worries @TheMarvelFan ! Good luck with your project; I think this one is ready to go as is; I've asked colleagues for a quick review before a merge given the changes.
